src/rest_vol_dataset.c

```diff
@@ -1060,6 +1085,8 @@ RV_dataset_write(size_t count, void *dset[], hid_t mem_type_id[], hid_t _mem_spa
                    transfer_info[i].u.write_info.base64_encoded_values);
 #endif
 
+            if (transfer_info[i].u.write_info.write_body)
+                RV_free(transfer_info[i].u.write_info.write_body);
```

Comment:
It would be good to set this pointer to NULL after freeing it so that if we fail to allocate it again below for whatever reason, the error cleanup code won't accidentally double-free the buffer.

Reply:
I have implemented this suggestion.